gobblin-compaction/src/main/java/gobblin/compaction/count/EventCountFromFileName.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.count;
+
+import java.util.List;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Implementation of {@link HdfsEventCountProvider} gets event count from a file name. File name should have the count in the past part, spearated by ".".
+ * Files written using {@link gobblin.writer.AvroHdfsTimePartitionedWithRecordCountsWriter} will follow the pattern.
+ * E.g., part.1.123.avro, the event count will be 123.
+ */
+@Slf4j
+public class EventCountFromFileName implements HdfsEventCountProvider {
+  @Override
+  public long getEventCount(FileSystem fs, List<Path> paths) {
+    long eventCount = 0;
+    try {
+      for (Path path : paths) {
+          String[] parts = FilenameUtils.removeExtension(path.getName()).split("\\.");
+          eventCount += Long.parseLong(parts[parts.length - 1]);
```

Comment:
Instead of requiring the record count to be the last component (which isn't the case for compaction output files), one way to make it more flexible is to create an interface `FilenameFormat` with method `constructFilename(String... components)`, and create a subinterface `FilenameWithRecordCountFormat` with another method `getRecordCount(String fileName)`. Then pass an instance to this method.

Also please rename `event` to `record` in non-Kafka-specific classes since event is a concept specific to Kafka, and record is more general.